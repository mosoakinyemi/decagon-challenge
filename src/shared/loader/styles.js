import {StyleSheet} from 'react-native';
import {fonts} from '../../fonts';

export const styles = StyleSheet.create({
  text: {
    ...fonts.boldText18,
    textAlign: 'center',
  },
  loader: {
    backgroundColor: 'rgba(255,255,255,0.7)',
    position: 'absolute',
    flex: 1,
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 100,
  },
});
