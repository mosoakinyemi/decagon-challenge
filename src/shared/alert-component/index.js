import React, {useState} from 'react';
import {View, Text, Modal} from 'react-native';
import {styles} from './styles';

export const AlertComponent = ({text}) => {
  return (
    // <Modal animationType="slide" transparent={true} visible={true}>
    <View style={styles.container}>
      <Text style={styles.text}>{text}</Text>
    </View>
    // </Modal>
  );
};
