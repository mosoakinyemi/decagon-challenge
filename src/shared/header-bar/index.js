import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import Icon from 'react-native-vector-icons/Entypo';
import {hp} from '../resposive-dimension';

const HeaderBar = ({title, onRightPress, onLeftPress}) => {
  return (
    <View style={styles.container}>
      {onLeftPress && (
        <TouchableOpacity style={styles.sideIcon} onPress={onLeftPress}>
          <Icon name="chevron-thin-left" size={hp(16)} />
          <Text style={styles.backText}>Back</Text>
        </TouchableOpacity>
      )}
      <Text style={styles.title}>{title}</Text>
      <TouchableOpacity style={styles.sideIcon} onPress={onRightPress}>
        {onRightPress && <Text style={styles.backText}>Home</Text>}
      </TouchableOpacity>
    </View>
  );
};

export default HeaderBar;
