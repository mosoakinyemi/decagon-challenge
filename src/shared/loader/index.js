import React from 'react';
import {ActivityIndicator, View, Text} from 'react-native';
import {colors} from '../../colors';
import {styles} from './styles';

export const Loader = ({text}) => {
  return (
    <View style={styles.loader}>
      <ActivityIndicator size="large" color={colors.dark} />
      <Text style={styles.text}>{text}</Text>
    </View>
  );
};
