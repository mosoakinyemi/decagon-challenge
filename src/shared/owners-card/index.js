import React, {PureComponent} from 'react';
import {View, Text} from 'react-native';
import {styles} from './styles';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {wp} from '../resposive-dimension';

export class OwnersCard extends PureComponent {
  render() {
    const {
      bio,
      car_color,
      car_model,
      car_model_year,
      gender,
      job_title,
      email,
      first_name,
      last_name,
    } = this.props.data;
    return (
      <View style={styles.card}>
        <Text style={styles.name}>
          {first_name} {last_name} ({gender})
        </Text>

        <View style={styles.containerRow}>
          <View>
            <View style={styles.row}>
              <View style={styles.row}>
                <MaterialIcons name="work" size={wp(15)} style={styles.icon} />
                <Text style={styles.detail}>{job_title}</Text>
              </View>

              <MaterialIcons name="email" size={wp(15)} style={styles.icon} />
              <Text style={styles.detail}>{email}</Text>
            </View>

            <View style={styles.row}>
              <View style={styles.row}>
                <MaterialIcons
                  name="directions-car"
                  size={wp(15)}
                  style={styles.icon}
                />
                <Text style={styles.detail}>
                  {car_model} {car_model_year}
                </Text>
              </View>
              <MaterialIcons
                name="color-lens"
                size={wp(15)}
                style={styles.icon}
              />
              <Text style={styles.detail}>{car_color}</Text>
            </View>
          </View>

          <View style={styles.bioSection}>
            <Text style={styles.bio}>Bio:</Text>
            <Text style={styles.bioDetail}>{bio}</Text>
          </View>
        </View>
      </View>
    );
  }
}
