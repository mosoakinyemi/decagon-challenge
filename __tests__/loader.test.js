import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import {Loader} from '../src/shared/loader';

jest.useFakeTimers();

describe('<Loader/>', () => {
  it('Loader component renders correctly', () => {
    const text = 'Incorrect Path';
    const instance = renderer.create(<Loader text={text} />);
    expect(instance).toMatchSnapshot();
  });

  it('Loader component renders correctly after an update', () => {
    const value1 = 'Hello';
    const value2 = 'Hi';

    const instance = renderer.create(<Loader text={value1} />);
    const textInstance = instance.root.findByType(Text);
    expect(textInstance.props.children).toBe(value1);

    instance.update(<Loader text={value2} />);
    expect(textInstance.props.children).toBe(value2);
  });
});
