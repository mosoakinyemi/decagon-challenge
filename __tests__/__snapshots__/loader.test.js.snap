// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> Loader component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "rgba(255,255,255,0.7)",
      "bottom": 0,
      "flex": 1,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 100,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2f3138"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Object {
        "color": "#2f3138",
        "fontSize": 29.5,
        "fontWeight": "bold",
        "lineHeight": 44.5,
        "textAlign": "center",
      }
    }
  >
    Incorrect Path
  </Text>
</View>
`;
